data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/RemotePeerForwarder.java

```diff
@@ -156,6 +156,7 @@ private Map<String, List<Record<Event>>> groupRecordsBasedOnIdentificationKeys(
             }
             if (numMissingIdentificationKeys == identificationKeys.size()) {
                 recordsMissingIdentificationKeys.increment(1);
+                identificationKeyValues.clear();
```

Comment:
Please add a test case for this behavior.

Reply:
The second test case test for recordsMissingIdentificationKeys to be incremented. Do you want the test to check for some thing else too?